tests/lib/cbprintf_package/src/main.c

```diff
@@ -923,7 +923,7 @@ ZTEST(cbprintf_package, test_cbprintf_package_convert_static)
 	uint32_t copy_flags = CBPRINTF_PACKAGE_CONVERT_RW_STR;
 
 	clen = cbprintf_package_convert(spackage, slen, NULL, 0, copy_flags, NULL, 0);
-	zassert_true(clen == slen + sizeof(test_str1) + 1/*null*/ - 2 /* arg+ro idx gone*/);
+	zassert_true(clen == slen + (int)sizeof(test_str1) + 1 /*null*/ - 2 /* arg+ro idx gone*/);
```

Comment:
This cast looks suspicious to me, its a string length comparison but I guess we need signed casting because... why?

Reply:
there's no single solution I suppose, Yuval proposed a different one here https://github.com/zephyrproject-rtos/zephyr/pull/78614/files#diff-cd2a686f83bf01b5ca4a68be63cb5c470f0413e7609411bba864eeb8e25fe885R927, I don't particularly care, just pulled this from a PR that was already approved, feel free to suggest any alternative that you think make more sense, I suppose it's just a matter of making the types coherent

Reply:
cool, picked Yuval version and recredited the commit